### PR TITLE
Fix the code used to get the working directory

### PR DESCRIPTION
**Pull request by ashvindeodhar**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: cns/configuration/configuration.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ func ReadConfig() (CNSConfig, error) {
 	// Check if env set for config path otherwise use default path
 	configpath, found := os.LookupEnv("CNS_CONFIGURATION_PATH")
 	if !found {
-		dir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+		dir, err := os.Getwd()
```
Comment (tamilmani1989) [inline cns/configuration/configuration.go]: It was stated that "Getwd returns a rooted path name corresponding to the current directory". So if we run the exe from different directory, will this work?

File: cns/configuration/configuration.go [hunk 2]
```diff
@@ -46,7 +46,22 @@ func ReadConfig() (CNSConfig, error) {
 	// Check if env set for config path otherwise use default path
 	configpath, found := os.LookupEnv("CNS_CONFIGURATION_PATH")
 	if !found {
-		dir, err := filepath.Abs(filepath.Dir(os.Args[0]))
+		var (
+			dir string
+			ex  string
+			err error
+		)
+		ex, err = os.Executable()
```
Comment (tamilmani1989) [inline cns/configuration/configuration.go]: can we make this as common function..

File: cns/configuration/configuration.go [hunk 3]
```diff
@@ -46,7 +46,22 @@ func ReadConfig() (CNSConfig, error) {
...
+		if err == nil {
+			dir = filepath.Dir(ex)
```
Comment (tamilmani1989) [inline cns/configuration/configuration.go]: is this test both in linux and windows?

File: cns/configuration/configuration.go (hunk 2 above)
Comment (ashvindeodhar) [reply cns/configuration/configuration.go]: done

File: cns/configuration/configuration.go (hunk 3 above)
Comment (ashvindeodhar) [reply cns/configuration/configuration.go]: yes

File: cns/configuration/configuration.go [hunk 4]
```diff
@@ -56,7 +57,7 @@ func ReadConfig() (CNSConfig, error) {
 		//dir + string(os.PathSeparator) + defaultConfigName
```
Comment (tamilmani1989) [inline cns/configuration/configuration.go]: its not your change...can you please remove this?

File: common/utils.go [hunk 5]
```diff
@@ -323,3 +324,23 @@ func GetAzureCloud(url string) (string, error) {
...
+
+func GetExecutableDirectory() (string, error) {
+	var (
+		dir string
+		ex  string
+		err error
+	)
+	ex, err = os.Executable()
+	if err == nil {
+		dir = filepath.Dir(ex)
```
Comment (tamilmani1989) [inline common/utils.go]: can you add a comment on why are we doing filepath.dir on nil case and EvalSymlinks on err case

File: cns/networkcontainers/networkcontainers_windows.go [hunk 6]
```diff
@@ -250,12 +250,13 @@ func getAzureNetworkContainerBinaryPath() (string, error) {
 		err        error
 	)
 
-	if workingDir, err = filepath.Abs(filepath.Dir(os.Args[0])); err != nil {
+	workingDir, err = common.GetExecutableDirectory()
+	if err != nil {
 		return binaryPath,
 			fmt.Errorf("[Azure CNS] Unable to find working directory. Error: %v. Cannot continue", err)
```
Comment (tamilmani1989) [inline cns/networkcontainers/networkcontainers_windows.go]: make sure we are logging this somewhere

File: cns/configuration/configuration.go (hunk 4 above)
Comment (ashvindeodhar) [reply cns/configuration/configuration.go]: done

File: cns/networkcontainers/networkcontainers_windows.go (hunk 6 above)
Comment (ashvindeodhar) [reply cns/networkcontainers/networkcontainers_windows.go]: this error bubbles up and gets printed

File: common/utils.go (hunk 5 above)
Comment (ashvindeodhar) [reply common/utils.go]: done

**Pull request merged**